x/rewards/keeper/providers.go

```diff
@@ -0,0 +1,123 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/rewards/types"
+	subscriptionTypes "github.com/lavanet/lava/x/subscription/types"
+)
+
+func (k Keeper) AggregateRewards(ctx sdk.Context, provider, chainid string, adjustmentDenom uint64, rewards math.Int) {
+	index := types.BasePayIndex{Provider: provider, ChainID: chainid}
+	basepay, found := k.getBasePay(ctx, index)
+	adjustedPay := sdk.NewDecFromInt(rewards).QuoInt64(int64(adjustmentDenom))
```

Comment:
let's do a redundant check here that adjustedPay is smaller eq than rewards

Reply:
done